nifi-commons/nifi-repository-encryption/src/main/java/org/apache/nifi/repository/encryption/AesGcmByteArrayRepositoryEncryptor.java

```diff
@@ -74,9 +73,17 @@ protected byte[] encrypt(final byte[] record, final String recordId, final Strin
         try {
             final byte[] encryptedRecord = cipher.doFinal(record);
             final byte[] serializedMetadata = getMetadata(keyId, cipher.getIV(), encryptedRecord.length);
-            return Arrays.concatenate(serializedMetadata, encryptedRecord);
+            return concatenate(serializedMetadata, encryptedRecord);
```

Comment:
Maybe use `org.apache.commons.lang3.ArrayUtils.addAll()` here, to avoid the need for a helper method?

Reply:
I considered that, but this approach avoids introducing an additional dependency on commons-lang3 for this library, so that's the reason for the manual approach.